File: Computing_algorithms/lab_02/Point.py
class Point:
    def __init__(self, x, y, derivative, second_derivative) -> None:
        self.x = x
        self.y = y
        self.derivative = derivative
        self.second_derivative = second_derivative

    def __init__(self, x, y) -> None:
        self.x = x
        self.y = y
        self.derivative = None
        self.second_derivative = None

File: Computer_graphics/lab_06/point.cpp
#include "point.h"

Point::Point()
{
    this->_x = 0;
    this->_y = 0;
}

Point::Point(int x, int y)
{
    this->_x = x;
    this->_y = y;
}

Point::Point(const Point &point)
{
    this->_x = point.x();
    this->_y = point.y();
}

Point &Point::operator=(const Point &point)
{
    this->_x = point.x();
    this->_y = point.y();

    return *this;
}

Point &Point::operator=(const Point &&point)
{
    this->_x = point.x();
    this->_y = point.y();

    return *this;
}

int Point::x() const
{
    return this->_x;
}

int Point::y() const
{
    return this->_y;
}
